cloudinit/net/sysconfig.py

```diff
@@ -1021,6 +1022,16 @@ def render_network_state(
         if self.netrules_path:
             netrules_content = self._render_persistent_net(network_state)
             netrules_path = subp.target_path(target, self.netrules_path)
+            if os.path.exists(netrules_path):
+                # There should be a more elegant way to get the path
```

Comment:
The `Paths` class has `.get_runpath("sem")`, but I don't think we have access to the `Paths` object in the renderers currently.

To use this we'll need to pass the distros' `self._paths` value to the renderer instantiation (`render_cls()`) in `_get_renderer()` under `cloudinit/distros/__init__.py`.

That will require a new arg in the renderer classes, but I imagine that other renderers will probably need this as well someday, so I think that this would be reasonable to do.